openpype/pipeline/publish/lib.py

```diff
@@ -864,6 +864,109 @@ def _validate_transient_template(project_name, template_name, anatomy):
                          ).format(template_name, project_name))
 
 
+def get_published_workfile_instance(context):
+    """Find workfile instance in context"""
+    for i in context:
+        is_workfile = (
+            "workfile" in i.data.get("families", []) or
+            i.data["family"] == "workfile"
+        )
+        if not is_workfile:
+            continue
+
+        # test if there is instance of workfile waiting
+        # to be published.
+        if not i.data.get("publish", True):
+            continue
+
+        return i
+
+
+def replace_published_scene(instance, replace_in_path=True):
```

Comment:
The naming of this makes it feel like it's about to replace a published scene file instead of swapping out a workfile entry for a published workfile entry.

Reply:
Will fix all others comments in https://github.com/ynput/OpenPype/pull/5191 , I don't want to touch this PR unnecessary as  there is conflict hell as it is.